nativelib/src/main/scala/scala/scalanative/runtime/ExecutionContext.scala

```diff
@@ -25,4 +25,17 @@ object ExecutionContext {
       }
     }
   }
+
+  private[runtime] def loopRunOnce(): Int = {
+    if (queue.nonEmpty) {
+      val runnable = queue.remove(0)
+      try {
+        runnable.run()
+      } catch {
+        case t: Throwable =>
+          QueueExecutionContext.reportFailure(t)
+      }
+    }
+    queue.size
```

Comment:
What do you think about a one line comment suggesting that `queue.size`
has a limited duration of validity? Anyone thinking about it will understand
that it is valid only until the next Future is created but who is creating Futures
and when can be quite opaque. 

The `queue.size` is known valid just after the method return but can go invalid
at the next method call. I believe the Scala substitution principle is designed
to make it opaque how a given statement is implemented.

Another way of stating this, perhaps clearer, is that a non-zero return will  remain
true until the next loop() or loopRunOnce() but that a zero return becomes more
unreliable the older it gets.

Reply:
@LeeTibbert dedicated `loopRunOnce` allows to inject SN's loop into another event loop that is built with `libuv` for example.

A duplicated PR from me https://github.com/scala-native/scala-native/pull/2225 needed for this feature.